### PR TITLE
Function Default Parameters are underspecified

### PR DESCRIPTION
**Pull request by WalterBright**
Default arguments were incompletely specified.

### CODE REVIEW

**Comment by aG0aep6G** [comment]
The commit message shouldn't say that this fixes issue 19645. DMD has to implement the new spec before the issue can be closed.

File: spec/function.dd [hunk 1]
```diff
@@ -1349,9 +1349,41 @@ void foo(int x, int y = 3)
 foo(4);   // same as foo(4, 3);
 ---
 
-        $(P Default parameters are evaluated in the context of the
+        $(P Default parameters are resolved and semantically checked in the context of the
         function declaration.
-        If the default value for a parameter is given, all following
+---
```
Comment (aG0aep6G) [inline spec/function.dd]: Close the paragraph before starting a code section. HTML doesn't like `<pre>` in `<p>`.

Comment (WalterBright) [reply spec/function.dd]: ok

**Comment by thewilsonator** [comment]
@WalterBright you need to change the commit message, thats what the bot looks for.

**Comment by aG0aep6G** [comment]
@andralex, you've merged it with the bad commit message. I've cleaned up after you and reopened the issue.

**Comment by thewilsonator** [comment]
@aG0aep6G Thanks!

**Comment by andralex** [comment]
@aG0aep6G sorry, I assumed it'd been fixed already. Thanks for the fix.

**Pull request merged**